src/Common/src/System/Security/Cryptography/DerEncoder.cs

```diff
@@ -274,13 +351,129 @@ internal static byte[][] SegmentedEncodeNamedBitList(byte[] bigEndianBytes, int
+                throw new CryptographicException();
+            if (oidValue[1] != '.')
+                throw new CryptographicException();
+
+            int firstRid;
+
+            switch (oidValue[0])
+            {
+                case '0':
+                    firstRid = 0;
+                    break;
+                case '1':
+                    firstRid = 1;
+                    break;
+                case '2':
+                    firstRid = 2;
+                    break;
+                default:
+                    throw new CryptographicException();
+            }
+
+            int startPos = 2;
+            int nextDot = oidValue.IndexOf('.', startPos);
+
+            BigInteger rid = ParseOidRid(oidValue, startPos, nextDot);
+            rid += 40 * firstRid;
+            startPos = nextDot + 1;
+
+            // The worst case is "1.1.1.1.1", which takes 4 bytes (5 rids, with the first two condensed)
+            // Longer numbers get smaller: "2.1.127" is only 2 bytes. (81d (0x51) and 127 (0x7F))
+            // So length / 2 should prevent any reallocations.
+            List<byte> encodedBytes = new List<byte>(oidValue.Length / 2);
+
+            EncodeRid(encodedBytes, ref rid);
+
+            while (startPos > 0)
+            {
+                nextDot = oidValue.IndexOf('.', startPos);
+                rid = ParseOidRid(oidValue, startPos, nextDot);
+                startPos = nextDot + 1;
+
+                EncodeRid(encodedBytes, ref rid);
+            }
+
+            return new byte[][]
+            {
+                new byte[] { (byte)DerSequenceReader.DerTag.ObjectIdentifier }, 
+                EncodeLength(encodedBytes.Count),
+                encodedBytes.ToArray(),
+            };
+        }
+
+        /// <summary>
+        /// Make a constructed SEQUENCE of the byte-triplets of the contents, but leave
+        /// the value in a segmented form (to be included in a larger SEQUENCE).
+        /// </summary>
+        /// <param name="items">Series of Tag-Length-Value triplets to build into one sequence.</param>
+        /// <returns>The encoded segments { tag, length, value }</returns>
+        internal static byte[][] ConstructSegmentedSequence(params byte[][][] items)
+        {
+            Debug.Assert(items != null);
+
+            byte[] data = ConcatenateArrays(items);
+
+            return new byte[][]
+            {
+                new byte[] { ConstructedSequenceTag }, 
+                EncodeLength(data.Length),
+                data,
+            };
+        }
+
         /// <summary>
         /// Make a constructed SEQUENCE of the byte-triplets of the contents.
         /// Each byte[][] should be a byte[][3] of {tag (1 byte), length (1-5 bytes), payload (variable)}.
         /// </summary>
         internal static byte[] ConstructSequence(params byte[][][] items)
         {
+            return ConstructSequence((IEnumerable<byte[][]>)items);
```

Comment:
`internal static byte[] ConstructSequence(params byte[][][] items)` is a better match than `internal static byte[] ConstructSequence(IEnumerable<byte[][]> items)` without the cast, preferring a StackOverflowException to something functional :)